proto/text.go

```diff
@@ -420,7 +420,7 @@ func (tm *TextMarshaler) writeStruct(w *textWriter, sv reflect.Value) error {
 				// in writeAny.
 				if fv.Kind() == reflect.Ptr && fv.IsNil() {
 					// Use errors.New so writeAny won't render quotes.
-					msg := errors.New("/* nil */")
+					msg := errors.New("# nil")
```

Comment:
This is actually a bug.  As shown in the testcase, the output is not a valid textproto, i.e. field name without value.  V2 will output an empty message for this as the oneof has been set to one of the fields.  I'd prefer to follow the same as v2.

Reply:
I agree. I think returning any `error` that contains syntactically important parts of the output format is being overly clever, especially using an `error` type to avoid rendering quotes elsewhere.

Basically, the code as was is confusing, which is why it has a comment explaining itself. Since we’re here, we could instead make it less confusing, and much simpler.